### PR TITLE
Gh 25/add rtl preview functions

### LINKED ISSUE #25: 🌐 Add RTL Preview Functions for More Components
### Description
Currently, only a few components have RTL (Right-to-Left) preview functions to test layout behavior for languages like Arabic and Hebrew. We need to add RTL previews for several components that have directional elements (text alignment, icon positioning, leading/trailing icons) to ensure proper RTL support.

### Background
The codebase already has examples of RTL previews:
- `BottomSheetHeader` uses `CompositionLocalProvider(LocalLayoutDirection provides LayoutDirection.Rtl)`
- `CountdownTimer` uses `@Preview(locale = "ar-AR")` annotation

#### 1. **KPCheckbox** with text/content
**File:** `core/src/main/java/com/trendyol/design/core/checkbox/CheckBox.kt`

**Current state:** Only has previews with `CheckboxContainerType.None`

**Needed:**
```kotlin
@Preview(showBackground = true)
@Composable
private fun CheckboxWithTextRTLPreview() {
    PreviewTheme {
        CompositionLocalProvider(LocalLayoutDirection provides LayoutDirection.Rtl) {
            KPCheckbox(
                style = KPCheckboxStyle.Primary,
                size = KPCheckboxSize.Medium,
                containerType = CheckboxContainerType.Text("تحديد الخيار"),
                checked = true,
                onCheckedChange = { },
            )
        }
    }
}
```

#### 2. **KPRadioButton** with text/content
**File:** `core/src/main/java/com/trendyol/design/core/radiobutton/TrendyolRadioButton.kt`

**Needed:** RTL previews showing radio buttons with `RadioButtonContainerType.Text` and `RadioButtonContainerType.Content` to verify text/content alignment next to the radio button

#### 3. **KPToolbar**
**File:** `core/src/main/java/com/trendyol/design/core/toolbar/Toolbar.kt`

**Needed:** RTL previews for all toolbar styles:
- `KPToolbarStyle.Icon.EndIcon` - verify icon flips to correct side
- `KPToolbarStyle.TextButton` - verify button alignment
- `KPToolbarStyle.Icon.EndTwoIcons` - verify two icons position correctly
- With title text in Arabic/Hebrew to verify text alignment

#### 4. **KPSearchBar**
**File:** `core/src/main/java/com/trendyol/design/core/search/SearchBar.kt`

**Current state:** Has 8 preview functions, none for RTL

**Needed:** RTL previews showing:
- Leading search icon positioning
- Trailing cancel icon positioning (when text is present)
- Placeholder text alignment
- Input text alignment

**Example:**
```kotlin
@Preview(showBackground = true)
@Composable
private fun MediumFilledSearchBarRTLPreview() {
    PreviewTheme {
        CompositionLocalProvider(LocalLayoutDirection provides LayoutDirection.Rtl) {
            KPSearchBar(
                modifier = Modifier.padding(8.dp),
                value = "بحث",
                style = KPSearchBarStyle.Filled,
                size = KPSearchBarSize.Medium,
                onValueChange = {},
                onTrailingIconClick = {},
            )
        }
    }
}
```

#### 5. **KPInfoBox** with icon positioning
**File:** `core/src/main/java/com/trendyol/design/core/infobox/KPInfoBox.kt`

**Current state:** Has previews with `iconPosition = Arrangement.Start` and `Arrangement.End`, but no RTL versions

**Needed:** RTL previews to verify:
- Icon position flips correctly (Start becomes End, End becomes Start in RTL)
- Text alignment is correct
- Border rendering in RTL

**Example:**
```kotlin
@Preview(showBackground = true)
@Composable
private fun OrangeInfoBoxRTLPreview() {
    PreviewTheme {
        CompositionLocalProvider(LocalLayoutDirection provides LayoutDirection.Rtl) {
            KPInfoBox(
                style = KPInfoBoxStyle.Orange,
                text = "لوريم إيبسوم دولور سيت أميت، كونسيكتيتور أديبيسينغ",
                iconPosition = Arrangement.Start,
            )
        }
    }
}
```

#### 6. **Input fields with leading/trailing icons**
**Files:** `core/src/main/java/com/trendyol/design/core/inputfield/`

**Needed:** RTL previews for text fields showing proper icon positioning and text alignment

### Acceptance Criteria
- [ ] Add at least one RTL preview for each component listed above
- [ ] Use appropriate RTL text samples (Arabic/Hebrew) for preview content
- [ ] Verify that the preview demonstrates the directional behavior clearly
- [ ] Follow existing RTL preview patterns in the codebase
- [ ] Ensure screenshot tests still pass (if applicable)

### Implementation Notes
Use one of these approaches:
1. Wrap content with `CompositionLocalProvider(LocalLayoutDirection provides LayoutDirection.Rtl)`
2. Use `@Preview(locale = "ar-AR")` annotation

Both approaches are valid; use whichever is more consistent with the component file's existing style.

### PR DESCRIPTION
**Pull request by burkido**
### What does this PR do?
Complete the issue #25. New RTLs added in provided informations.

## 📋 Related Issue
Closes #25 
Related to #25 

### How has this been tested?

- [x] Screenshot tests
- [ ] UI tests
- [ ] Manual testing

### Test Coverage

- [x] Main functionality
- [ ] Edge cases
- [ ] Different Android versions
- [ ] Different screen sizes/densities
- [ ] RTL layout support
- [ ] Accessibility features

### General
- [x] Code follows project coding standards
- [x] Self-review completed
- [x] No debugging code left behind
- [x] Documentation updated if needed

### Testing
- [x] Existing tests still pass
- [ ] New tests added if applicable
- [ ] Screenshot tests updated if UI changed
- [ ] Edge cases covered

### Breaking Changes
- [ ] This change introduces breaking changes
- [x] This change is backward compatible

## 🔄 Migration Required

- [x] No migration required
- [ ] Migration required (explain below)

**Migration Details:**

## 📸 Screenshot Test Changes

- [x] Screenshot tests were not affected
- [ ] Screenshot tests were updated
- [ ] New screenshot tests were added

**Screenshot Changes Explained:**

---

### Bug Fix Testing

- [ ] Bug reproduction scenario
- [ ] Edge cases specific to the bug
- [ ] Regression testing

### Bug Fix Screenshots/Videos

**Before:**

**After:**

---

### Components/Screens Covered

- [ ] Component 1
- [ ] Component 2

### Test Scenarios Added

- [ ] Default state
- [ ] Loading state
- [ ] Error state
- [ ] Success state
- [ ] Empty state
- [ ] Focused state
- [ ] Disabled state
- [ ] Different content lengths
- [ ] Interaction states

### Screenshot Test Coverage

- [ ] Different screen densities (mdpi, hdpi, xhdpi, xxhdpi)
- [ ] Different screen sizes (phone, tablet)
- [ ] RTL layout
- [ ] Different font scales
- [ ] Accessibility features enabled

### Screenshot Test Commands

```bash
# Generate new screenshots
./gradlew executeScreenshotTests -Precord

# Verify screenshots
./gradlew executeScreenshotTests
```

### Screenshot Quality Checklist
- [ ] Screenshots are clear and properly framed
- [ ] All relevant UI elements are visible
- [ ] Screenshots represent realistic usage scenarios
- [ ] Consistent with existing screenshot style
- [ ] No unnecessary screenshot files added

### Coverage Improvement

- [ ] Adds coverage for previously untested component
- [ ] Improves coverage for existing component
- [ ] Adds regression prevention for specific bug
- [ ] Covers new component states/variations

---

### CODE REVIEW

File: core/src/main/java/com/trendyol/design/core/checkbox/CheckBox.kt [hunk 1]
```diff
@@ -182,3 +185,35 @@ private fun TrendyolXSmallCheckboxPreviewDisabled() {
         )
     }
 }
+
+@Preview(showBackground = true)
+@Composable
+private fun TrendyolMediumCheckboxWithTextRTLPreview() {
+    PreviewTheme {
+        CompositionLocalProvider(LocalLayoutDirection provides LayoutDirection.Rtl) {
```
Comment (turkergoksu) [inline core/src/main/java/com/trendyol/design/core/checkbox/CheckBox.kt]: Maybe we could create a `PreviewRTLTheme` or add a parameter to `PreviewTheme` named `isRTL`

**Review by turkergoksu — DISMISSED** [review]

**Review by iiremdogan — APPROVED** [review]
thanks for your efforts 👏

**Review by turkergoksu — APPROVED** [review]
